nameko/messaging.py

```diff
@@ -434,13 +180,46 @@ def handle_message(self, body):
         """
         self.queue = queue
         self.requeue_on_error = requeue_on_error
-        super(Consumer, self).__init__(**kwargs)
+        self.consumer_options = consumer_options
+        # TODO it's bad that we eat all the keyword arguments as consumer opts
+        super(Consumer, self).__init__()
+
+    @property
+    def amqp_uri(self):
+        return self.container.config[AMQP_URI_CONFIG_KEY]
 
     def setup(self):
-        self.queue_consumer.register_provider(self)
+        verify_amqp_uri(self.amqp_uri)
+
+        config = self.container.config
+
+        heartbeat = self.consumer_options.pop(
+            'heartbeat', config.get(HEARTBEAT_CONFIG_KEY, DEFAULT_HEARTBEAT)
+        )
+        prefetch_count = self.consumer_options.pop(
+            'prefetch_count', config.get(
+                PREFETCH_COUNT_CONFIG_KEY, DEFAULT_PREFETCH_COUNT
+            )
+        )
+        accept = self.consumer_options.pop(
+            'accept', serialization.setup(config)[1]
+        )
```

Comment:
We now allow overrides of these "consumer options" by passing them to the constructor of this entrypoint